packages/core/src/errors.ts

```diff
@@ -31,6 +34,16 @@ export class ProviderError<InnerError = unknown> extends ComposableError<InnerEr
   }
 }
 
+export class HandleOwnerChangeError extends CustomError {
+  constructor(
+    public handle: Handle,
+    public expectedAddress: Cardano.PaymentAddress,
+    public actualAddress: Cardano.PaymentAddress | null
+  ) {
+    super('Results are different between expectedAddress and actualAddress for the handle.');
```

Comment:
```suggestion
    super(`Expected: ${expectedAddress} for handle $${handle}. Actual: ${actualAddress}`);
```

Reply:
aa8096909c8ee4b312f8d2d9395b28766ebd4ff2 should solve this